src/tray/cocoa/SDL_tray.m

```diff
@@ -276,13 +276,21 @@ void SDL_SetTrayTooltip(SDL_Tray *tray, const char *tooltip)
     return entry->submenu;
 }
 
-const SDL_TrayEntry **SDL_GetTrayEntries(SDL_TrayMenu *menu, int *size)
+SDL_TrayEntry **SDL_GetTrayEntries(SDL_TrayMenu *menu, int *size)
 {
+    SDL_TrayEntry **out = SDL_malloc(sizeof(SDL_TrayEntry *) * menu->nEntries);
+
+    if (!out) {
+        return NULL;
+    }
+
     if (size) {
-        *size = menu->nEntries;
+        *size = (int) menu->nEntries;
```

Comment:
Why isn't nEntries int?

Reply:
Originally, it was because nEntries couldn't be negative and it avoided casts when mixing it with memory/addressing-related functions. I did expose the size parameters as int in the interface though, so I can change it to int if necessary. (Alternatively, I can change the parameters to be size_t, but I'm not sure if size_t is an acceptable type to have in the API)